docs/src/main/asciidoc/jfr.adoc

```diff
@@ -0,0 +1,364 @@
+
+image::jfr-event-browser.png[alt=JDK Mission Control Event Browser view,role="center"]
+
+To see Quarkus events in chronological order on a thread, select the `Java application` and `Threads` on the left side of the JMC.
+
+image::jfr-java-ap-thread.png[alt=JDK Mission Control thread view,role="center"]
+
+The standard configuration does not show Quarkus events.
+We have to do three tasks to see the Quarkus events.
+
+1. Right-click and select `Edit Thread Activity Lanes...`.
+2. Select the plus button to add a new lane on the left side, then check to display that lane.
+3. Select Quarkus as the event type that lane will display and press OK.
+
+image::jfr-edit-thread-activity-lanes.png[alt=JDK Mission Control Edit Thread Activity Lanes,role="center"]
+
+Now we can see the Quarkus events per thread.
+
+image::jfr-thread.png[alt=JDK Mission Control thread view,role="center"]
+
+[NOTE]
+====
+Non-blocking is where multiple processes are processed apparently simultaneously in a single thread.
+Therefore, this extension records multiple JFR events concurrently, and a number of events might overlap on the JMC.
+This could make it difficult for you to see the events you want to see.
+To avoid this, we recommend to use xref:#identifying-requests[Request ID] to filter events so that you only see the information about the requests you want to see.
+====
+
+== Events
+
+=== Identifying Requests
+This extension works with the OpenTelemetry extension.
+The events recorded by this extension have a trace ID and a span ID. These are recorded with the OpenTelemetry IDs respectively.
+
+This means that after we identify the trace and span IDs of interest from the UI provided by the OpenTelemetry implementation, we can immediately jump to the details in JFR using those IDs.
+
+If we have not enabled the OpenTelemetry extension, this extension creates an ID for each request and links it to JFR events as a traceId.
+In this case, the span ID will be null.
+
+For now, Quarkus only has REST events, but we plan to use this ID to link each event to each other as we add more events in the future.
+
+=== Event Implementation Policy
+When JFR starts recording an event, the event does not record to JFR yet, but when it commits that event, the event is recorded.
+Therefore, events that have started recording at dump time but have not yet been committed are not dumped.
+This is unavoidable due to the design of JFR.
+This means that events are not recorded forever if there are prolonged processing.
+Therefore, you will not be aware of prolonged processing.
+
+To solve this problem, Quarkus can also record start and end events at the beginning and end of processing.
+These events are disabled by default.
+However, we can enable these events on JFR.(described below)
+
+=== REST API Event
+This event is recorded when either `resteasy-reactive` or `resteasy-classic` extension is enabled.
```

Comment:
The documentation still uses the old naming here for `resteasy-reactive`

Reply:
Thank you. These have been fixed. I may have forgotten to change them when I squashed these